static/js/common/upload-addon.js

```diff
@@ -431,6 +431,20 @@
                       $("<p>").text(gettext("Your submission will be automatically signed.")).appendTo(upload_results);
                     }
 
+                    if (results.validation.is_upgrade_to_webextension) {
+                        var warning_box = $('<div>').attr('class', 'important-warning').appendTo(upload_results);
+
+                        $('<h5>').text(gettext("WebExtension upgrade")).appendTo(warning_box);
+                        $('<p>').text(gettext(
+                            "We allow and encourage an upgrade but you cannot reverse this process. Once your users have the WebExtension installed, they will not be able to install a legacy add-on."
+                        )).appendTo(warning_box);
+
+                        $('<a>').text(gettext('Porting a legacy Firefox add-on on MDN'))
+                                .attr('href', 'https://developer.mozilla.org/Add-ons/WebExtensions/Porting_a_legacy_Firefox_add-on')
+                                .attr('target', '_blank')
```

Comment:
If we really must have `_blank` then we should add the relevant rel attributes to prevent leaking the opener object. That'll be `rel="noopener noreferrer"`. 

Note: `noreferrer` is needed since FF currently doesn't support `noopener` yet.